src/Symfony/Component/DependencyInjection/Compiler/CheckServiceClassExistsPass.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\IncorrectServiceClassException;
+use Symfony\Component\DependencyInjection\Reference;
+
+class CheckServiceClassExistsPass implements CompilerPassInterface
```

Comment:
This should probably be implemented as a recursive pass, so that all definitions that got inlined during the optimization phase are also checked (otherwise, you would have to run it earlier, but it is harder to avoid issues for unused services that gets removed, which is often a valid usage of non-existent classes (related to optional dependencies of bundles for instance)